lib/Rfc1035StubResolver.php

```diff
@@ -88,8 +94,25 @@ public function resolve(string $name, int $typeRestriction = null): Promise
         }
 
         return call(function () use ($name, $typeRestriction) {
-            if (!$this->config) {
-                yield $this->reloadConfig();
+            if (!$this->configStatus) {
+                try {
+                    yield $this->reloadConfig();
+                } catch (ConfigException $e) {
+                    $this->configStatus = self::CONFIG_FAILED;
```

Comment:
This should probably be inside `reloadConfig`?

Reply:
done